cmd/ctl/pkg/renew/renew.go

```diff
@@ -0,0 +1,249 @@
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+	apiutil "github.com/jetstack/cert-manager/pkg/api/util"
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+)
+
+// Options is a struct to support renew command
+type Options struct {
+	// The Namespace that the Certificate to be renewed resided in
+	Namespace string
+
+	LabelSelector string
+
+	All  bool
+	Wait bool
+
+	AllNamespaces bool
+
+	PollTime time.Duration
+	Timeout  time.Duration
+
+	genericclioptions.IOStreams
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		IOStreams: ioStreams,
+	}
+}
+
+// NewCmdRenew returns a cobra command for renewing Certificates
+func NewCmdRenew(ioStreams genericclioptions.IOStreams, factory cmdutil.Factory) *cobra.Command {
+	o := NewOptions(ioStreams)
+
+	cmd := &cobra.Command{
+		Use:   "renew",
+		Short: "Mark a Certificate for manual renewal",
+		Long:  "Mark a Certificate for manual renewal",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(factory, cmd, args))
+			cmdutil.CheckErr(o.Validate(cmd, args))
+			cmdutil.CheckErr(o.Run(factory, cmd, args))
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.LabelSelector, "selector", "l", o.LabelSelector, "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
+	cmd.Flags().BoolVarP(&o.AllNamespaces, "all-namespaces", "A", o.AllNamespaces, "If present, wait for Certificates across all namespaces to become ready. Namespace in current context is ignored even if specified with --namespace.")
+	cmd.Flags().BoolVar(&o.All, "all", o.All, "Renew all Certificates in the given Namespace, or all namespaces with --all-namespaces enabled.")
+	cmd.Flags().BoolVarP(&o.Wait, "wait", "w", o.Wait, "Wait for all Certificates to become ready once being marked for renewal.")
+	cmd.Flags().DurationVar(&o.PollTime, "poll-time", time.Second*2, "Poll period between checking Certificates to become ready. Used in conjunction with --wait.")
+	cmd.Flags().DurationVar(&o.Timeout, "timeout", 0, "The length of time to wait before ending watch, zero means never. Any other values should contain a corresponding time unit (e.g. 1s, 2m, 3h).")
+
+	return cmd
+}
+
+// Validate validates the provided options
+func (o *Options) Validate(cmd *cobra.Command, args []string) error {
+	if len(o.LabelSelector) > 0 && len(args) > 0 {
+		return errors.New("cannot specify Certificate arguments as well as label selectors")
+	}
+
+	if o.All && len(args) > 0 {
+		return errors.New("cannot specify Certificate arguments as well as --all flag")
```

Comment:
`cannot specify Certificate names as well as the --all flag` (names instead of arguments). Also, perhaps `in conjunction with`? Not sure if the language is overly complex though.